rotkehlchen/chain/ethereum/modules/curve/decoder.py

```diff
@@ -502,13 +505,32 @@ def _decode_curve_events(self, context: DecoderContext) -> DecodingOutput:
         return DEFAULT_DECODING_OUTPUT
 
     def _decode_curve_gauge_events(self, context: DecoderContext) -> DecodingOutput:
-        if context.tx_log.topics[0] not in (GAUGE_DEPOSIT, GAUGE_WITHDRAW):
+        if context.tx_log.topics[0] not in (GAUGE_DEPOSIT, GAUGE_WITHDRAW, GAUGE_VOTE):
             return DEFAULT_DECODING_OUTPUT
 
-        provider = hex_or_bytes_to_address(context.tx_log.topics[1])
+        found_event_modifying_balances = False
         gauge_address = context.tx_log.address
+        if context.tx_log.topics[0] == GAUGE_VOTE:
```

Comment:
Why do you have this in the same function. This makes no sense. Make a separate function only for gauge votes. There is no reason to mix those 2 functions and have all those ifs.